js/main.js

```diff
@@ -166,3 +171,36 @@ function renderEditForm(entry) {
   $inputTitle.value = entry.title;
   $inputNotes.value = entry.notes;
 }
+$form.addEventListener('click', handleDeleteClickEvent);
+
+function handleDeleteClickEvent(event) {
+  if (event.target.matches('a')) {
+    $modal.classList.remove('hidden');
+  }
+}
+
+$modal.addEventListener('click', handleModalBtnsClickEvent);
+
+function handleModalBtnsClickEvent(event) {
+  var $anchorDataViewAttribute = event.target.getAttribute('data-view');
+  if (event.target === $cancelBtn) {
+    viewSwap($anchorDataViewAttribute);
+    $modal.classList.add('hidden');
+  }
+  if (event.target === $confirmBtn) {
+    var $deleteCurrentLi = document.querySelector('[data-entry-id="' + data.editing.id + '"]');
+    var indexOfEntries = data.entries.indexOf(data.editing);
+    data.entries.splice(indexOfEntries, 1);
+    $deleteCurrentLi.remove();
+    data.editing = null;
+    viewSwap($anchorDataViewAttribute);
+    $modal.classList.add('hidden');
+    $h1NewEntry.textContent = 'New Entry';
+    $placeHolderImage.setAttribute('src', 'images/placeholder-image-square.jpg');
+    $form.reset();
+    if (data.entries.length === 0) {
+      $divElementNoEntries.classList.remove('hidden');
+      data.nextEntryId = 1;
```

Comment:
You definitely should not ever reset `nextEntryId`. Even if all prior entries have been deleted, we still want to use fully unique ids.